### ISSUE TITLE
Overlay/Backdrop to disable interaction with all but a single _existing_ component

### ISSUE THREAD

**Issue by jaamison** [palantir/blueprint#2653]
#### Feature request

__tl;dr:__ A tool to "modalize" a single arbitrary component on a page, in-place (without (re-)rendering it in a Portal)

I recently came across a case that requires me to conditionally and temporarily make a component modal. The component displays read-only data, but can enter an "edit mode" during which it is swapped with an editor component with identical dimensions & position and very similar layout, but with text inputs instead of display-only data. An in-place editor - no big deal so far. However, I want the editor component to 1) know when a user clicks/interacts outside the component (onBlur and event.relatedTarget won't cut it for reasons I can explain later), and 2) have the option of disabling all interaction with the rest of the page.

This sounds like a job for a traditional blurprint Overlay or a Popover, but rendering the editor in a Portal and specifying its position manually via coordinates so it exactly covers the original component seems kludgey. I also have a lot of these components (50+) on a given page.

I was thinking an overlay-like component that prevents interaction with the entire page except a small portion of it, effectively isolating a single component as the only portion of the page that allows interaction. This would allow me to temporarily make any single component "modal" while leaving its component instance intact and in its current place in the component hierarchy.

I'm still trying to decide if it's worth it to build a solution for this, or if I just tell our design team to take a hike back to the drawing board. However, if this is something that others may find useful I'd be more inclined to invest the time. Is this functionality something that the blueprint/palantir team would be at all interested in internally? I'm definitely willing to do my own heavy lifting - I just don't want to move forward completely alone if I'm the only person who really wants this. Thoughts?

**Comment by giladgray** [comment]
@jaamison sorry, we do not typically take component requests. this should be quite possible with a conditional Overlay or Dialog and you can easily disable portals with `usePortal={false}`.
```jsx
if (inModal) {
    return <Dialog isOpen={true} usePortal={false}>{contents}</Dialog>;
} else {
    return contents;
}
```

**Comment by jaamison** [comment]
Thanks @giladgray, I understand completely.

Purely FYI: usePortal={false} won't work for me because I need the whole page covered, not just the container. My dilemma was how to overlay the entire page without using a Portal. I actually ended up coming up with a different solution that accomplishes this.

Good luck with the v3 launch!
